### ISSUE TITLE
Sibling transfer issues warning if multiapps are on different execute_ons

### ISSUE THREAD

**Issue by zachmprince** [idaholab/moose#22668]
## Bug Description

When performing a sub-app to sub-app transfer, i.e. sibling transfer, where the multiapps have different `execute_on`s, it is impossible to avoid a warning stating that the transfer does not have the same `execute_on` flags as either multiapp. It is my opinion that the transfer should execute during the `to_multi_app` execution since sibling transfers are before multiapp execution. And there should only be a warning if it doesn't match the `to_multi_app` 

## Steps to Reproduce

Here is the main and sub app inputs recreating the issue:

main.i:
```
[MultiApps]
  [sub1]
    type = FullSolveMultiApp
    input_files = sibling_sub.i
    execute_on = TIMESTEP_BEGIN
    cli_args = 'Variables/u/initial_condition=2'
  []
  [sub2]
    type = FullSolveMultiApp
    input_files = sibling_sub.i
    execute_on = TIMESTEP_END
  []
[]

[Transfers]
  [sibling_transfer]
    type = MultiAppCopyTransfer
    from_multi_app = sub1
    to_multi_app = sub2
    source_variable = u
    variable = u
  []
[]


[Mesh]
  [min]
    type = GeneratedMeshGenerator
    dim = 1
    nx = 1
  []
[]

[Problem]
  solve = false
  kernel_coverage_check = false
  skip_nl_system_check = true
  verbose_multiapps = true
[]

[Executioner]
  type = Steady
[]
```

sibling_sub.i:
```
[Variables/u]
[]

[Postprocessors/avg_u]
  type = ElementAverageValue
  variable = u
[]


[Mesh]
  [min]
    type = GeneratedMeshGenerator
    dim = 1
    nx = 1
  []
[]

[Problem]
  solve = false
  kernel_coverage_check = false
  skip_nl_system_check = true
  verbose_multiapps = true
[]

[Executioner]
  type = Steady
[]
```

Without specifying the transfer's `execute_on` it seems to want to execute on every possible exec flag (`INITIAL`, `TIMESTEP_BEGIN`,  `TIMESTEP_END`, `FINAL`) and it issues the warning:
```
*** Warning ***
The following warning occurred in the object "sibling_transfer", of type "MultiAppCopyTransfer".

MultiAppTransfer execute_on flags do not match associated to_multi_app execute_on flags
```

With `Transfers/sibling_transfer/execute_on=TIMESTEP_END` it again executes on every flag and issues the warning:
```
*** Warning ***
The following warning occurred in the object "sibling_transfer", of type "MultiAppCopyTransfer".

MultiAppTransfer execute_on flags do not match associated from_multi_app execute_on flags
```

Finally, with `Transfers/sibling_transfer/execute_on='TIMESTEP_BEGIN TIMESTEP_END'` it executes on every flag and issues the warning:
```
*** Warning ***
The following warning occurred in the object "sibling_transfer", of type "MultiAppCopyTransfer".

MultiAppTransfer execute_on flags do not match associated from_multi_app execute_on flags


*** Warning ***
The following warning occurred in the object "sibling_transfer", of type "MultiAppCopyTransfer".

MultiAppTransfer execute_on flags do not match associated to_multi_app execute_on flags
```

To summarize:
- Transfer wants to execute on every flag, no matter what.
- No `execute_on`: warning for `to_multi_app`
- `execute_on=TIMESTEP_END`: warning for `from_multi_app`
- `execute_on='TIMESTEP_BEGIN TIMESTEP_END'`: warning for both `from_multi_app` and `to_multi_app`

## Impact

Two major impacts:
1. Although it does not affect the answer, performing transfers on every exec flag could be costly for some transfers.
2. The unavoidable warning message is erroneous (IMO) and a test with this type of transfer will always fail because of the warning.

**Comment by YaqiWang** [comment]
I guess that sibling transfer was developed for enabling Gauss-Seidel style multiapp coupling, like `subapp1->subapp2->mainapp` instead of the ordering of two subapps on the same execute flag is undetermined. If the subapps are on different execute flag, they do not need to use sibling transfer. But having this sibling transfer on different execute flags can save some lines in the inputs and should be supported I think.

**Comment by GiudGiud** [comment]
We can support this. 
The zealous warning message was added later to prevent user mistakes. I ll address this soon

**Comment by zachmprince** [comment]
The sibling transfer also helps with the fact that the main app does not need to have the same mesh. My execution resembles `subapp1->mainapp->subapp2`.

**Comment by YaqiWang** [comment]
Yeah, good point. If the two apps are having the same mesh but main app does not, the direct transfer can use copy transfers (might having issues if different partitionings happen on the two subapps).

**Comment by GiudGiud** [comment]
the generalFieldTransfers will be able to handle that case within the context of siblings transfers
I ll get back on that work after ANS
